lib/puppet/resource/type.rb

```diff
@@ -370,12 +367,27 @@ def set_resource_parameters(resource, scope)
       scope[TITLE] = resource.title
       scope[NAME] =  resource.name
     end
-
     scope.class_set(self.name,scope) if hostclass? || node?
 
-    assign_defaults(resource, scope, resource_hash)
-    validate_resource_hash(resource, resource_hash)
-    resource_hash.each { |param, value| exceptwrap { scope[param] = value }}
+    begin
+      elevel = scope.ephemeral_level
+      param_scope = scope.new_parameter_scope(arguments.keys)
+
+      # Assign directly to the parameter scope to avoid scope parameter validation at this point. It
+      # will happen anyway when the values are assigned to the scope after the parameter scoped has
+      # been popped.
```

Comment:
Actually, it produces an error 'Illegal numeric variable name' but using `"Y${0}es"` instead of `"Y$0es"` works OK though. I'll add that test.